daemon/src/main/scala/io/gearpump/cluster/worker/Worker.scala

```diff
@@ -303,14 +304,29 @@ private[cluster] object Worker {
     procLauncher: ExecutorProcessLauncher) extends Actor {
     import launch.{appId, executorId, resource}
 
-    val config = {
-      val systemConfig = context.system.settings.config
+    val executorConfig: Config = {
+      val workerConfig = context.system.settings.config
 
-      // try to resolve user config
-      val userConfig = Option(launch.executorJvmConfig).flatMap{ jvmConfig =>
+      val submissionConfig = Option(launch.executorJvmConfig).flatMap{ jvmConfig =>
         Option(jvmConfig.executorAkkaConfig)
       }.getOrElse(ConfigFactory.empty())
-      userConfig.withFallback(systemConfig)
+
+      resolveExecutorConfig(workerConfig, submissionConfig)
+    }
+
+    // For some config, worker has priority, for others, user Application submission config
+    // have priorities.
+    private def resolveExecutorConfig(workerConfig: Config, submissionConfig: Config): Config = {
+      val config = submissionConfig.withoutPath(Constants.GEARPUMP_HOSTNAME)
```

Comment:
The code is good. But from user perspective, better to print a warning about what config items have been ignored.